### PR TITLE
feat: universal links + app links for mobile app rollout

### LINKED ISSUE #923: Mobile app rollout on web
Add universal links and maybe announcement banner

### PR DESCRIPTION
**Pull request by malinmalliyawadu**
Closes #923

Makes links to `volunteers.everybodyeats.nz` open the Everybody Eats mobile app when installed (iOS Universal Links + Android App Links), and adds an iOS Safari Smart App Banner. The homepage already promotes the app, so this covers the "universal links" half of the issue end-to-end.

## Web (`web/`)
- **apple-app-site-association** — added an `applinks` section (kept existing `webcredentials`). Claims `/shifts`, `/shifts/*`, `/dashboard`, `/profile*`, `/friends`, `/friends/*`, `/achievements*` only; auth/api/survey paths stay in the browser.
- **assetlinks.json** — new Android App Links file (package `com.everybodyeats.app`).
- **next.config.ts** — `headers()` serving both files as `application/json` (the AASA file has no extension, so iOS would otherwise reject it).
- **layout.tsx** — `apple-itunes-app` Smart App Banner (app-id `6760931588`).

## Mobile (`mobile/`)
- **app.json** — `applinks:volunteers.everybodyeats.nz` added to iOS `associatedDomains`; new `autoVerify` Android HTTPS intent filter scoped to the same paths.
- **app/+native-intent.ts** — rewrites incoming web URLs to mobile routes (web `/shifts/:id` → `/shift/[id]`, `/friends/:id` → `/user/[id]`, etc.), mirroring `notification-routing.ts`. Handles the `/shifts/mine` and `/shifts/details` edge cases and passes OAuth redirects through untouched.

## Docs
- `web/docs/universal-links.md` — setup, the synced path list, and verification steps.

## ⚠️ Required before Android release
`assetlinks.json` ships with a **placeholder** SHA-256 fingerprint (`REPLACE_WITH_ANDROID_APP_SIGNING_SHA256_FINGERPRINT`). Android links won't verify until it holds the real **app-signing** cert from Play Console → App signing (or `eas credentials`). iOS works without it.

## Notes
- The mobile `app.json` / `associatedDomains` changes only take effect in a **new native build** (not existing installs or OTA updates).
- All JSON validated; the deep-link path mapper was tested against a full case suite.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by malinmalliyawadu** [comment]
Thanks for the review — addressed the substantive items in `582f5957`:

**Addressed**
- **#1 Open redirect / #3 URL validation / #6 logging** — Extracted the mapper into `mobile/lib/deep-link-routing.ts`. It now only honours http(s) URLs for our own host (`volunteers.everybodyeats.nz`), collapses any non-internal path to `/`, and never returns an embedded absolute URL — while still passing custom-scheme OAuth redirects through untouched. `+native-intent.ts` is now a thin wrapper that logs on failure. (Note: native deep-link routing isn't a true web open-redirect since the OS only delivers our associated-domain URLs, but the hardening is worth having.)
- **#2 Android `pathPrefix` over-matching** — Android intent filters now use exact `path` + trailing-slash `pathPrefix` (e.g. `/shifts` and `/shifts/`), so iOS and Android claim the identical URL set and `/shifts-archive`-style paths no longer match.
- **#5 Test coverage** — Added `mobile/lib/deep-link-routing.test.ts` (19 cases incl. the `/shifts/mine` edge case, query params, full URLs, OAuth passthrough, and the malicious-URL cases). Full mobile suite green (72/72).
- **#10 Docs** — Added a troubleshooting section (iOS AASA caching, `adb` app-link verification, wrong-screen mapping).

**Intentionally not changed**
- **#4 CI gate on the placeholder fingerprint** — A hard gate would block *this* PR, since the placeholder is deliberately present until the Android release (the real app-signing SHA-256 isn't available yet). It's documented as a pre-release step instead.
- **#7 unused `initial`** — Not actually bound as a variable; it's kept in the type for signature clarity.
- **#8 / #9 path-sync automation & conditional banner** — Skipped as scope creep. The three path lists are documented as the sync point, and a site-wide Smart App Banner is standard (it's a no-op outside iOS Safari).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: web/public/.well-known/assetlinks.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[
+  {
+    "relation": ["delegate_permission/common.handle_all_urls"],
+    "target": {
+      "namespace": "android_app",
+      "package_name": "com.everybodyeats.app",
+      "sha256_cert_fingerprints": [
+        "REPLACE_WITH_ANDROID_APP_SIGNING_SHA256_FINGERPRINT"
```
Comment (malinmalliyawadu) [inline web/public/.well-known/assetlinks.json]:
```suggestion
        "B5:68:7A:3C:9E:FF:D7:5C:E7:2F:E0:02:9C:30:93:22:EC:A6:C4:78:74:D0:F4:20:26:B1:6A:96:FF:96:0A:D3"
```
